src/main/java/com/google/step/youtube/PromoCode.java

```diff
@@ -9,12 +9,14 @@
 @AutoValue
 public abstract class PromoCode {
 
-    public static PromoCode create(String promoCode, String videoId, String videoTitle, Date videoUploadDate) {
-        return new AutoValue_PromoCode(promoCode, videoId, videoTitle, videoUploadDate);
+    public static PromoCode create (String promoCode, String snippet, String videoId, String videoTitle, Date videoUploadDate) {
```

Comment:
optional: you have 5 arguments all with a primitive type `String`.  it can be easy to mix up the argument order and it would still compile.  Consider using https://github.com/google/auto/blob/master/value/userguide/builders.md
(less necessary if the arguments were of different types, b/c the compiler would complain.)
might help  with readability, too.

if you do, do in a subsequent CL.  not this one.

Reply:
Added issue to icebox.